.travis.yml

```diff
@@ -2,18 +2,37 @@ language: node_js
 node_js:
   - "10"
 
-env:
-  - OCAML=4.02.3
-  - OCAML=4.02.3+buckle-master
-  - OCAML=4.03.0
-  - OCAML=4.04.2
-  - OCAML=4.05.0
-  - OCAML=4.06.1
-  - OCAML=4.07.0
-  - ESY_BUILD=YES ESY__CACHE=/home/travis/.esy
+matrix:
+  include:
+    - os: osx
+      env: OCAML=4.07.0
+    - os: osx
+      env: ESY_BUILD=YES ESY__CACHE=/home/travis/.esy
+    - os: windows
+      env: OCAML=4.07.0
+    - os: windows
+      env: ESY_BUILD=YES ESY__CACHE=/home/travis/.esy
+    - os: linux
+      env: OCAML=4.02.3
+    - os: linux
+      env: OCAML=4.02.3+buckle-master
+    - os: linux
+      env: OCAML=4.03.0
+    - os: linux
+      env: OCAML=4.04.2
+    - os: linux
+      env: OCAML=4.05.0
+    - os: linux
+      env: OCAML=4.06.1
+    - os: linux
+      env: OCAML=4.07.0
+    - os : linux
+      env: ESY_BUILD=YES ESY__CACHE=/home/travis/.esy
 
 cache:
   directories:
+  - /usr/local/Cellar/opam
+  - /usr/local/Cellar/tidy-html5
   - /home/travis/.opam
```

Comment:
macOS users are under something like `/Users/travis` IIRC. I think you may be able to get this working on both Linux and macOS with `$HOME/.opam`, etc., instead. Not sure about Windows yet.

Alternatively, only for opam, you can try a local opam switch. But that won't help with `.esy` and `.nvm`, so it's probably not worth it.